packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNavButton/useTeachingPopoverCarouselNavButton.tsx

```diff
@@ -24,23 +24,27 @@ export const useTeachingPopoverCarouselNavButton_unstable = (
 ): TeachingPopoverCarouselNavButtonState => {
   const { index, onClick, as = 'a' } = props;
   const appearance = usePopoverContext_unstable(context => context.appearance);
-  const setCurrentPage = useTeachingPopoverCarouselContext_unstable(context => context.setCurrentPage);
-  const currentPage = useTeachingPopoverCarouselContext_unstable(context => context.currentPage);
-  const totalPages = useTeachingPopoverCarouselContext_unstable(context => context.totalPages);
-  const onPageChange = useTeachingPopoverCarouselContext_unstable(context => context.onPageChange);
+
+  const setCurrentPage = useCarouselContext_unstable(context => context.setIndex);
+  const currentPage = useCarouselContext_unstable(context => context.currentIndex);
+  const totalPages = useCarouselContext_unstable(context => context.totalPages);
+  const onPageChange = useCarouselContext_unstable(context => context.onPageChange);
+  const store = useCarouselContext_unstable(context => context.store);
   const isSelected = currentPage === index;
 
+  const values = store.getSnapshot();
+
   const setNewPage = React.useCallback(
     event => {
       if (onClick) {
         onClick(event);
       }
       if (!event.defaultPrevented && isHTMLElement(event.target)) {
         setCurrentPage(index);
-        onPageChange?.(event, { event, type: 'click', currentPage: index });
+        onPageChange?.(event, { event, type: 'click', index, value: values[index] });
       }
     },
-    [onClick, setCurrentPage, index, onPageChange],
+    [onClick, setCurrentPage, index, onPageChange, values],
```

Comment:
We should try to avoid subscription as much as possible, for example there we can just report `index`, we can get `value` in another place